Remotion/Core/UnitTests/Context/SafeContextTests/SafeContextTest.Task.ContinueWith.cs

```diff
@@ -0,0 +1,367 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Remotion.Context;
+using static Remotion.UnitTests.Context.SafeContextTests.SafeContextTaskTest.ContinueWithTestFlags;
+
+namespace Remotion.UnitTests.Context.SafeContextTests
+{
+  public partial class SafeContextTaskTest
+  {
+    [Flags]
+    public enum ContinueWithTestFlags
+    {
+      None = 0,
+      HasState = 1,
```

Comment:
I can handle those bitflags, so no need to chenge this now. I did however find a nice way to make those more maintainable by using the bit-shift syntax (`1 << 2`, `1 << 3`, etc)

Reply:
The other option would be `0b0001`, `0b0010`, ... although I find this to be less readable overall.

Reply:
yeah, bits are annoying. the bitshift variant as the advantage that it's the easiest to sanity check when you get to longer flag lists. I guess the sensible cutoff is at 1024 the latest.